File: data/multilang_comparison/program_julia/src/func1.c
#include "program_julia.h" 
array_t* func1(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array225;
   if (pCounter > 0) {
      array225 = vars->data[--pCounter];
      array225->refC++;
      DEBUG_COPY(array225->id);
   } else {
      array225 = (array_t*)malloc(sizeof(array_t));
      array225->size = 137;
      array225->refC = 1;
      array225->id = 225;
      array225->data = (unsigned int*)malloc(array225->size*sizeof(unsigned int));
      memset(array225->data, 0, array225->size*sizeof(unsigned int));
      DEBUG_NEW(array225->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array225;
   array_t* array226 = func3(&params0, loopsFactor);
   DEBUG_RETURN(array226->id);
   free(params0.data);
   if(PATH0 & 1) {
      array_t* array229;
      if (pCounter > 0) {
         array229 = vars->data[--pCounter];
         array229->refC++;
         DEBUG_COPY(array229->id);
      } else {
         array229 = (array_t*)malloc(sizeof(array_t));
         array229->size = 963;
         array229->refC = 1;
         array229->id = 229;
         array229->data = (unsigned int*)malloc(array229->size*sizeof(unsigned int));
         memset(array229->data, 0, array229->size*sizeof(unsigned int));
         DEBUG_NEW(array229->id);
      }
      unsigned int loop66 = 0;
      unsigned int loopLimit66 = (rand()%loopsFactor)/2 + 1;
      for(; loop66 < loopLimit66; loop66++) {
         for (int i = 0; i < array229->size; i++) {
            array229->data[i]--;
         }
         for (int i = 0; i < array226->size; i++) {
            if (array226->data[i] == 58) { 
               return array226;
            }
         }
      }
      array_t_param params1;
      params1.size = 3;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array225;
      params1.data[1] = array226;
      params1.data[2] = array229;
      array_t* array230 = func10(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array230->id);
      free(params1.data);
      array230->refC--;
      if(array230->refC == 0) {
         free(array230->data);
         free(array230);
         DEBUG_FREE(array230->id);
      }
      array229->refC--;
      if(array229->refC == 0) {
         free(array229->data);
         free(array229);
         DEBUG_FREE(array229->id);
      }
   }
   else {
      array_t_param params1;
      params1.size = 2;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array225;
      params1.data[1] = array226;
      array_t* array231 = func4(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array231->id);
      free(params1.data);
      array231->refC--;
      if(array231->refC == 0) {
         free(array231->data);
         free(array231);
         DEBUG_FREE(array231->id);
      }
   }
   array225->refC--;
   if(array225->refC == 0) {
      free(array225->data);
      free(array225);
      DEBUG_FREE(array225->id);
   }
   return array226;
}

File: data/multilang_comparison/program_julia/src/func1.jl
include("program_julia_head.jl")
function func1(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array193 = nothing
   if pCounter > 1
       array193 = vars.data[pCounter]
       pCounter -= 1
       array193.refC += 1
   else
       array193 = JArray(zeros(Int, 462), 462, 1, 193)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array193)
   params0.size += 1
   array194 = func3(params0, loopsFactor);
   if (PATH0 & 1) != 0
      array197 = nothing
      if pCounter > 1
          array197 = vars.data[pCounter]
          pCounter -= 1
          array197.refC += 1
      else
          array197 = JArray(zeros(Int, 951), 951, 1, 197)
      end
      loopLimit57 = (50)/2 + 1
      for i in 1:loopLimit57 
         for i in 0:(array193.size-1) 
            array193.data[i+1] -= 1
         end
         for i in 1:array194.size
             if array194.data[i] == 37
                     return array194
            end
         end
      end
      params1 = JArray_param{Int}()
      params1.size = 3
      push!(params1.data, array193)
      params1.size += 1
      push!(params1.data, array194)
      params1.size += 1
      push!(params1.data, array197)
      params1.size += 1
      array198 = func10(params1, get_path(), loopsFactor);
      array198.refC -= 1
      if array198.refC == 0
          empty!(array198.data)
          array198 = nothing
      end
      array197.refC -= 1
      if array197.refC == 0
          empty!(array197.data)
          array197 = nothing
      end
   
   else 
      params1 = JArray_param{Int}()
      params1.size = 2
      push!(params1.data, array193)
      params1.size += 1
      push!(params1.data, array194)
      params1.size += 1
      array199 = func4(params1, get_path(), loopsFactor);
      array199.refC -= 1
      if array199.refC == 0
          empty!(array199.data)
          array199 = nothing
      end
   end
   array193.refC -= 1
   if array193.refC == 0
       empty!(array193.data)
       array193 = nothing
   end
   return array194;
end

